src/SFA.DAS.ApplyService.InternalApi/Controllers/OrganisationSummaryController.cs

```diff
@@ -0,0 +1,296 @@
+    {
+        private readonly IInternalQnaApiClient _qnaApiClient;
+        private readonly IApplyRepository _applyRepository;
+        private readonly ILogger<OrganisationSummaryController> _logger;
+
+        public OrganisationSummaryController(IInternalQnaApiClient qnaApiClient, IApplyRepository applyRepository, ILogger<OrganisationSummaryController> logger)
+        {
+            _qnaApiClient = qnaApiClient;
+            _applyRepository = applyRepository;
+            _logger = logger;
+        } 
+
+        [HttpGet]
+        [Route("TypeOfOrganisation/{applicationId}")]
+        public async Task<IActionResult> GetTypeOfOrganisation(Guid applicationId)
+        {
+            _logger.LogInformation($"Retrieving type of organisation for application {applicationId}");
+            const string TRUE = "TRUE";
+                 var companyVerification = await
+                _qnaApiClient.GetQuestionTag(applicationId, RoatpWorkflowQuestionTags.UkrlpVerificationCompany);
+
+            var charityVerification = await
+                _qnaApiClient.GetQuestionTag(applicationId, RoatpWorkflowQuestionTags.UkrlpVerificationCharity);
+
+
+            if (companyVerification == TRUE && charityVerification == TRUE)
+            {
+                return Ok(GatewayOrganisationTypes.CompanyAndCharity);
+            }
+
+            if (companyVerification == TRUE)
+            {
+                return Ok(GatewayOrganisationTypes.Company);
+            }
+
+            if (charityVerification == TRUE)
+            {
+                return Ok(GatewayOrganisationTypes.Charity);
+            }
+
+            var soleTraderPartnership = await
+                _qnaApiClient.GetQuestionTag(applicationId, RoatpWorkflowQuestionTags.SoleTraderOrPartnership);
+
+            if (!string.IsNullOrEmpty(soleTraderPartnership))
+                return Ok(soleTraderPartnership);
+
+            return Ok(GatewayOrganisationTypes.StatutoryInstitute);
+        }
+
+        [HttpGet]
+        [Route(("DirectorData/Submitted/{applicationId}"))]
+        public async Task<IActionResult> GetDirectorsFromSubmitted(Guid applicationId)
+        {
+            _logger.LogInformation($"Retrieving submitted company directors for application {applicationId}");
+            var peopleInControl = new List<PersonInControl>();
+
+            var directorsData =
+                await _qnaApiClient.GetTabularDataByTag(applicationId,
+                    RoatpWorkflowQuestionTags.CompaniesHouseDirectors);
+
+            if (directorsData?.DataRows == null || !directorsData.DataRows.Any()) return Ok(peopleInControl);
+
+            foreach (var director in directorsData.DataRows.Where(x => x.Columns.Any()).OrderBy(x => x.Columns[0]))
+            {
+                var directorName = director.Columns[0];
+                var directorDob = string.Empty;
+                if (director.Columns.Any() && director.Columns.Count >= 2)
```

Comment:
If there aren't any columns line 84 would fail so not sure if this part of the check is necessary (or it could be in the wrong place?)

Reply:
There's a Where clause a few lines above that filters on Columns.Any()

Reply:
There are but 2 lines up would still fail if there are no columns wouldn't it? var directorName = director.Columns[0];

Reply:
There's an any in the foreach that checks there's always at least one

Reply:
Hi Mike, I can't seem to get a re-review requested for this in the github, and as you're sick and I've been asked to merge quite a few bits in, I'm going ahead and resolving this conversation so I can merge it.  I have double checked it all addresses your concerns though.